### ISSUE TITLE
Provider panics when generating plan

### ISSUE THREAD

**Issue by kian99** [juju/terraform-provider-juju#274]
### Description

I'm seeing the juju terraform provider panic when running `terraform apply` or `terraform plan`. Previously this worked fine and nothing in the model has changed. The controller was recently upgraded to Juju 2.9.44 but testing the provider on a different model with the same controller works fine. The output and panic details aren't too long so I will paste them in the relevant section below.

### Urgency

Casually reporting

### Terraform Juju Provider version

0.7.0, 0.8.0

### Terraform version

v1.5.5

### Terraform Configuration(s)

The Terraform plan can be found here - https://git.launchpad.net/~kian-parvin/canonical-terraform-plans/+git/canonical-terraform-plans/tree/livepatch-server/environments/stg-k8s?h=update-livepatch-spec

This folder contains the Terraform spec, it also calls in a module which is two directory up.

### Reproduce / Test

```shell
`terraform apply` or `terraform plan`
```

### Debug/Panic Output

```shell
Planning failed. Terraform encountered an error while generating this plan.

╷
│ Error: Plugin did not respond
│ 
│   with module.livepatch.juju_application.livepatch,
│   on ../../modules/livepatch-server/main.tf line 11, in resource "juju_application" "livepatch":
│   11: resource "juju_application" "livepatch" {
│ 
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ReadResource call. The plugin logs may contain more details.
╵
╷
│ Error: Plugin did not respond
│ 
│   with module.livepatch.juju_application.ingress,
│   on ../../modules/livepatch-server/main.tf line 95, in resource "juju_application" "ingress":
│   95: resource "juju_application" "ingress" {
│ 
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ReadResource call. The plugin logs may contain more details.
╵

Stack trace from the terraform-provider-juju_v0.8.0 plugin:

panic: interface conversion: interface {} is map[string]interface {}, not string

goroutine 61 [running]:
github.com/juju/terraform-provider-juju/internal/juju.ConfigEntryToString({0x1fd38a0?, 0xc0006a5c80?})
	github.com/juju/terraform-provider-juju/internal/juju/applications.go:86 +0x105
github.com/juju/terraform-provider-juju/internal/juju.(*ConfigEntry).String(...)
	github.com/juju/terraform-provider-juju/internal/juju/applications.go:72
github.com/juju/terraform-provider-juju/internal/provider.resourceApplicationRead({0x2729d60?, 0xc0007695f0?}, 0xc000445b80, {0x1e5db60?, 0xc0005ff200})
	github.com/juju/terraform-provider-juju/internal/provider/resource_application.go:347 +0x113a
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0xc000198d20, {0x2729d60, 0xc0007695f0}, 0xd?, {0x1e5db60, 0xc0005ff200})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.26.1/helper/schema/resource.go:724 +0x12e
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc000198d20, {0x2729d60, 0xc0007695f0}, 0xc0006752b0, {0x1e5db60, 0xc0005ff200})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.26.1/helper/schema/resource.go:1015 +0x585
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadResource(0xc000604168, {0x2729d60?, 0xc0007694d0?}, 0xc000523ec0)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.26.1/helper/schema/grpc_provider.go:613 +0x4a5
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadResource(0xc0002ca640, {0x2729d60?, 0xc000768d20?}, 0xc000776540)
	github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/tf5server/server.go:748 +0x4b1
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler({0x222a9c0?, 0xc0002ca640}, {0x2729d60, 0xc000768d20}, 0xc000275ce0, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:349 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000138000, {0x27326e0, 0xc0004ccea0}, 0xc00077a000, 0xc00021c900, 0x383cb30, 0x0)
	google.golang.org/grpc@v1.53.0/server.go:1336 +0xd33
google.golang.org/grpc.(*Server).handleStream(0xc000138000, {0x27326e0, 0xc0004ccea0}, 0xc00077a000, 0x0)
	google.golang.org/grpc@v1.53.0/server.go:1704 +0xa36
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	google.golang.org/grpc@v1.53.0/server.go:965 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.53.0/server.go:963 +0x28a

Error: The terraform-provider-juju_v0.8.0 plugin crashed!
```
```


### Notes & References

_No response_

**Comment by hmlanigan** [comment]
Please update with a copy of the plan you are using.

**Comment by kian99** [comment]
@hmlanigan Updated to include a link to the Terraform plan

**Comment by hmlanigan** [comment]
This issue was encountered while working on migration of the application resource from using the terraform sdk to the terraform framework.  It will be fixed in 0.9.0

https://github.com/juju/terraform-provider-juju/pull/265

**Comment by hmlanigan** [comment]
0.9.0 is being released
